### PR TITLE
Subcube + BoundedPlane

### PR DESCRIPTION
**Pull request by achaikou**
Two preparations for the next PR which will try to restructure the way we manage attributes (successfully or not is a different question)

1. Subvolume -> Subcube as I will likely name another structure Subvolume.
2. BoundedPlane was needed for nicer handling of plane information as this Subvolume kinda will represent data between RegularSurfaces based on the same horizontal stuff.

Changes should however make sense even if next PR is never merged.

### CODE REVIEW

**Review by yngve793 — APPROVED** [review]
First commit is simply a renaming, looks fine. 
Boundaries are related to the plane and thus it makes sense to group them. 
Row, col, methods on BoundedPlane is a great contribution.

File: internal/core/regularsurface.cpp
[the remark this answers is not in the document]
Comment (yngve793) [reply internal/core/regularsurface.cpp:1]: Both ways are fine, I was mainly poking to check if it is was thought through. 
Just keep it as it is.

File: internal/core/regularsurface.hpp [hunk 1]
```diff
@@ -90,31 +124,21 @@ class RegularSurface{
...
-    Point to_cdp(
-        std::size_t i
-    ) const noexcept(false);
-
-    /* World coordinates -> grid position */
-    Point from_cdp(
-        Point point
-    ) const noexcept (false);
+    RegularSurface(
+        float* data,
```
[the remark this answers is not in the document]
Comment (yngve793) [reply internal/core/regularsurface.hpp:138]: For this one I disagree. RegularSurface could be used in a different way or context in the future, and thus this safeguard will be lost. It should be the class own responsibility to ensure that the data it is representing is consistent.

[the remark this answers is not in the document]
Comment (achaikou) [reply internal/core/regularsurface.hpp:138]: Totally agree with you in theory, but how would you achieve this in practice?
float* doesn't know about size of its own data. It's just a pointer. It's size is stored in row x col.
What do you want to compare with row x col then?

File: internal/core/regularsurface.cpp [hunk 2]
```diff
@@ -89,40 +93,40 @@ Point RegularSurface::to_cdp(
...
-    return this->m_plane.m_inverse_transformation * point;
+    return this->m_inverse_transformation * point;
 }
 
 std::pair<std::size_t, std::size_t> as_pair(std::size_t row, std::size_t col) {
     return std::pair<std::size_t, std::size_t>(row, col);
 }
 
 float &RegularSurface::operator[](std::size_t i) noexcept(false) {
-    if (i >= this->size())
```
[the remark this answers is not in the document]
Comment (yngve793) [reply internal/core/regularsurface.cpp:103]: For me index is more descriptive than i, but it is just a suggestion.

File: internal/core/regularsurface.hpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (yngve793) [reply internal/core/regularsurface.hpp:138]: The constructor of the class is a good place to perform a test and throw an exception. In this way we are certan that any instance of the class is consistent.

[the remark this answers is not in the document]
Comment (achaikou) [reply internal/core/regularsurface.hpp:138]: So in theory I am all in for it, but as I still don't understand how it technically can be done, consider this improvement be one of the tasks you can do in the future and show clueless me the result :sweat_smile:

[the remark this answers is not in the document]
Comment (yngve793) [reply internal/core/regularsurface.hpp:138]: Sure, let put it up as an issue.

**Review by yngve793 — APPROVED** [review]
No further comments.

**Pull request merged**